### PR TITLE
new upsampler

### PR DESCRIPTION
**Pull request by shadowatyyy**

### CODE REVIEW

File: trax/models/research/funnel_transformer.py [hunk 1]
```diff
@@ -351,6 +362,8 @@ def FunnelTransformer(vocab_size,
                                   dropout_shared_axes, mode, ff_activation)
                     for _ in range(n_decoder_blocks)]
 
+  total_pool_size = pool_size * (len(encoder_segment_lengths) - 1)
```
Comment (syzymon) [inline trax/models/research/funnel_transformer.py]: should be `pool_size ** (len(encoder_segment_lengths) - 1)` ?

File: trax/models/research/funnel_transformer.py [hunk 2]
```diff
@@ -60,13 +62,23 @@ def SelectFirst():
...
 
 
-def _Upsample(short, long):
-  factor = -(-long.shape[1] // short.shape[1])  # ceil division
-  new_vecs = long + short.repeat(factor, axis=1)[:, :long.shape[1], :]
-  return new_vecs
+def _Upsampler(total_pool_size, separate_cls):
+  def _Upsample(short, long):
+    if separate_cls:
+      upsampled_short = jnp.append(
```
Comment (syzymon) [inline trax/models/research/funnel_transformer.py]: maybe `jnp.concatenate` here? in vanilla numpy `append` calls `concatenate` under the hood if `axis` is given, who knows how it is implemented in `jnp`, so I guess it's better to express it directly that you want to concatenate

Comment (shadowatyyy) [reply trax/models/research/funnel_transformer.py]: done

File: trax/models/research/funnel_transformer.py (hunk 1 above)
Comment (shadowatyyy) [reply trax/models/research/funnel_transformer.py]: done

**Review by syzymon — APPROVED** [review]
